src/components/MaterialUI/Inputs/DatePicker/TimePicker.js

```diff
@@ -0,0 +1,169 @@
+import React, { useState, useCallback } from "react";
+import styled from "styled-components";
+import { getThemeProp } from "../../../../utils";
+
+const TimeWrapper = styled.div`
+	margin: auto;
+`;
+
+const TimeZoneWrapper = styled.label`
+	margin-left: 5px;
+	font-family: ${getThemeProp(["fonts", "base"], "sans-serif")};
+	font-size: 12px;
+`;
+
+const TimePickerWrapper = styled.span`
+	display: flex;
+	flex-wrap: wrap;
+	width: 100%;
+`;
+
+const TimePickerSegmentWrapper = styled.select`
+	font-family: ${getThemeProp(["fonts", "base"], "sans-serif")};
+	font-size: 12px;
+	color: #333333;
+	margin-left: 0.5em;
+	margin-right: 3px;
+`;
+
+const hourOptions = [];
+for (var i = 1; i <= 12; i++) {
+	hourOptions.push({ value: i, label: i });
+}
+
+const minOptions = [
+	{ value: "00", label: "00" },
+	{ value: "15", label: "15" },
+	{ value: "30", label: "30" },
+	{ value: "45", label: "45" },
+];
+
+const ampmOptions = [
+	{ value: "AM", label: "AM" },
+	{ value: "PM", label: "PM" },
+];
+
+const parseTime = timeStr => {
+	var time = timeStr.match(/(\d+)(?::(\d\d))?\s*(p?)/i);
+	if (!time) {
+		return new Date();
+	}
+
+	let dt = new Date();
+	var hours = parseInt(time[1], 10);
+	if (time[3] === "p" && hours < 12) {
+		hours += 12;
+	}
+	dt.setHours(hours);
+	dt.setMinutes(parseInt(time[2], 10) || 0);
+	dt.setSeconds(0, 0);
+	return dt;
+};
+
+const leftFillNum = (num, targetLength) => num.toString().padStart(targetLength, 0);
+const isAM = time => time.getHours() < 12;
+const calculateHours = time => (time.getHours() === 0 || time.getHours() === 12 ? 12 : time.getHours() % 12);
+const calculateMins = time =>
+	time.getMinutes() >= 45 ? "45" : time.getMinutes() >= 30 ? "30" : time.getMinutes() >= 15 ? "15" : "00";
+const calculateAMPM = time => (isAM(time) ? "AM" : "PM");
```

Comment:
So, we don't support 24 hours clock, that is, 23:00 instead of 11:00PM ? I would be very surprised by that.

Reply:
Walter says to support it, so I am working on this now.